mesonbuild/interpreterbase/interpreterbase.py

```diff
@@ -281,9 +295,13 @@ def evaluate_if(self, node: mparser.IfClauseNode) -> T.Optional[Disabler]:
 
     def evaluate_comparison(self, node: mparser.ComparisonNode) -> InterpreterObject:
         val1 = self.evaluate_statement(node.left)
+        if val1 is None:
+            raise InvalidCodeOnVoid()
```

Comment:
```python
raise MesonException('Cannot compare a void statement on the left-hand side')
```
Seems like it would help people figure out better, what the issue is.